File: java/nec.unit-test/src/main/java/nec/Models/SeatBelt.java
package nec.Models;

public class SeatBelt {

    private ISeatBeltState _state;

    public SeatBelt() {
	this._state = () -> false;
    }

    public Boolean isFastened() {
	return _state.isFastened();
    }

    public void fasten() {
	this._state = () -> true;
    }

    public void unfasten() {
	this._state = () -> false;
    }
}

File: netCore/NEC.UnitTest/src/NEC.Models/SeatBelt.cs
using System;

namespace NEC.Models
{
    public class SeatBelt
    {
        private Func<bool> _state;

        public SeatBelt() => _state = () => false;

        public bool IsFastened() => _state();

        public void Fasten() => _state = () => true;

        public void Unfasten() => _state = () => false;
    }
}
